dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/ReferrerURLCookieHandler.java

```diff
@@ -141,19 +141,49 @@ public void invalidateReferrerURLCookie(HttpServletRequest req, HttpServletRespo
      */
     public void invalidateCookie(HttpServletRequest req, HttpServletResponse res, String cookieName, boolean enableHttpOnly) {
         Cookie c = new Cookie(cookieName, "");
+        Cookie existingCookie = getExistingCookieFromRequest(req, cookieName);
+        if (existingCookie != null) {
+            String path = existingCookie.getPath();
+            if (path != null) {
+                c.setPath(path);
+            }
+            String domain = existingCookie.getDomain();
+            if (domain != null) {
+                c.setDomain(domain);
+            }
+            c.setSecure(existingCookie.getSecure());
+            c.setHttpOnly(existingCookie.isHttpOnly());
```

Comment:
Match all of the available properties of the existing cookie to ensure the right cookie is invalidated.